dashboard/modules/reporter/reporter_head.py

```diff
@@ -112,6 +123,225 @@ async def get_cluster_status(self, req):
+
+        Args:
+            task_id (str): The ID of the task.
+            attempt_number (int): The attempt number of the task.
+
+        Returns:
+            tuple: A tuple containing the worker's PID (process ID) and IP address.
+
+        Raises:
+            ValueError: If the PID or IP is None, indicating that the information could not be fetched.
+            ValueError: If the task attempt is not in the "RUNNING" state.
+        """
+        option = ListApiOptions(
+            filters=[
+                ("task_id", "=", task_id),
+                ("attempt_number", "=", attempt_number),
+            ],
+            detail=True,
+            timeout=10,
+        )
+
+        result = await self._state_api.list_tasks(option=option)
+        tasks = result.result
+
+        pid = tasks[0]["worker_pid"]
+        logger.info(f"pid {type(pid)}: {pid}")
+
+        node_id = tasks[0]["node_id"]
+        ip = DataSource.node_id_to_ip[node_id]
+
+        state = tasks[0].get("state")
+        logger.info(f"ip {type(ip)}: {ip}")
+
+        if not pid or not ip:
+            raise ValueError(
+                f"pid or ip is None, could not fetch the info you need: pid = {pid}, ip = {ip}"
+            )
+        logger.info(f"node_id {type(node_id)}: {node_id}")
+
+        if state != "RUNNING":
+            raise ValueError(
+                f"The task attempt is not running: the current state is {state}."
+            )
+        logger.info(f"state {type(state)}: {state}")
+
+        return pid, ip
+
+    @routes.get("/task/traceback")
+    async def get_task_traceback(self, req) -> aiohttp.web.Response:
+        """
+        Retrieves the traceback information for a specific task.
+        We don't use pid and ip to get task cpu profile since ip and pid is bounded to a worker
+        and a worker may run different tasks at different time.
+        Therefore, we may provide wrong profile info since we don't know which task is running on the worker
+
+        Args:
+            req (aiohttp.web.Request): The HTTP request object.
+
+        Returns:
+            aiohttp.web.Response: The HTTP response containing the traceback information.
+
+        Raises:
+            ValueError: If the "task_id" parameter is missing in the request query.
+            ValueError: If the "attempt_number" parameter is missing in the request query.
+            ValueError: If the worker begins working on another task during the traceback retrieval.
+            aiohttp.web.HTTPInternalServerError: If there is an internal server error during the traceback retrieval.
+        """
+
+        if "task_id" not in req.query:
+            raise ValueError("task_id is required")
+        if "attempt_number" not in req.query:
+            raise ValueError("task's attempt number is required")
+
+        task_id = req.query.get("task_id")
+        attempt_number = req.query.get("attempt_number")
+
+        self.set_up_state_api()
```

Comment:
maybe we could set this up when the class being init?

Reply:
Good idea!